src/cursor/cur_layered.c

```diff
@@ -14,23 +14,20 @@ static int __clayered_open_cursors(WT_CURSOR_LAYERED *, bool);
 static int __clayered_reset_cursors(WT_CURSOR_LAYERED *, bool);
 static int __clayered_search_near(WT_CURSOR *cursor, int *exactp);
 
-/*
- * We need a tombstone to mark deleted records, and we use the special value below for that purpose.
- * We use two 0x14 (Device Control 4) bytes to minimize the likelihood of colliding with an
- * application-chosen encoding byte, if the application uses two leading DC4 byte for some reason,
- * we'll do a wasted data copy each time a new value is inserted into the object.
- */
-static const WT_ITEM __tombstone = {"\x14\x14", 3, NULL, 0, 0};
-
 /*
  * __clayered_deleted --
- *     Check whether the current value is a tombstone.
+ *     Check whether the current value is a tombstone in the layered cursor.
  */
 static WT_INLINE bool
-__clayered_deleted(const WT_ITEM *item)
+__clayered_deleted(WT_CURSOR_LAYERED *clayered, const WT_ITEM *item)
 {
-    return (item->size == __tombstone.size &&
-      memcmp(item->data, __tombstone.data, __tombstone.size) == 0);
+    if (clayered->ingest_cursor == NULL)
```

Comment:
This deserves an explanation of why the checks are here, or how tombstones are managed for layered cursors.

Reply:
added in [d8ddba3](https://github.com/wiredtiger/wiredtiger/pull/11621/commits/d8ddba30358aa6ce759248d211020290ca730d22)